source/slang/slang-emit-metal.cpp

```diff
@@ -253,6 +253,24 @@ bool MetalSourceEmitter::tryEmitInstStmtImpl(IRInst* inst)
     case kIROp_discard:
         m_writer->emit("discard_fragment();\n");
         return true;
+    case kIROp_MetalAtomicCast:
+    {
+        auto oldValName = getName(inst);
+        auto op0 = inst->getOperand(0);
+
+        m_writer->emit("atomic_");
+        emitType(op0->getDataType());
```

Comment:
This can work now, but it may subject to breakage in the future if the type has other modifiers like constexpr etc.

A more correct and future proof way is to introduce a IRMetalAtomicType(baseType) inst, and form a PtrType(IRMetalAtomicType(IntType)) as the result type of a MetalAtomicCast inst, and implement the emit logic for IRMetalAtomicType properly in the emit logic.

Reply:
That sounds good. I will try it tomorrow.